homeassistant/components/yale_smart_alarm/alarm_control_panel.py

```diff
@@ -46,66 +60,106 @@
-        return
+async def async_setup_entry(hass, entry, async_add_entities):
+    """Set up the alarm entry."""
 
-    add_entities([YaleAlarmDevice(name, client)], True)
+    async_add_entities(
+        [YaleAlarmDevice(coordinator=hass.data[DOMAIN][entry.entry_id]["coordinator"])]
+    )
 
 
-class YaleAlarmDevice(AlarmControlPanelEntity):
+class YaleAlarmDevice(CoordinatorEntity, AlarmControlPanelEntity):
     """Represent a Yale Smart Alarm."""
 
-    def __init__(self, name, client):
-        """Initialize the Yale Alarm Device."""
-        self._name = name
-        self._client = client
-        self._state = None
+    _state_map = {
+        YALE_STATE_DISARM: STATE_ALARM_DISARMED,
+        YALE_STATE_ARM_PARTIAL: STATE_ALARM_ARMED_HOME,
+        YALE_STATE_ARM_FULL: STATE_ALARM_ARMED_AWAY,
+    }
 
-        self._state_map = {
-            YALE_STATE_DISARM: STATE_ALARM_DISARMED,
-            YALE_STATE_ARM_PARTIAL: STATE_ALARM_ARMED_HOME,
-            YALE_STATE_ARM_FULL: STATE_ALARM_ARMED_AWAY,
-        }
+    _state = STATE_UNAVAILABLE
 
     @property
     def name(self):
         """Return the name of the device."""
-        return self._name
+        return self.coordinator.entry.data[CONF_NAME]
+
+    @property
+    def unique_id(self) -> str:
+        """Return the unique ID for this entity."""
+        return str(self.coordinator.entry.entry_id)  # type: ignore[attr-defined]
+
+    @property
+    def device_info(self) -> DeviceInfo:
+        """Return device information about this entity."""
+        return {
+            ATTR_NAME: self.coordinator.entry.data[CONF_NAME],  # type: ignore[attr-defined]
+            ATTR_MANUFACTURER: MANUFACTURER,
+            ATTR_MODEL: MODEL,
+            ATTR_IDENTIFIERS: {(DOMAIN, self.coordinator.entry.data[CONF_USERNAME])},  # type: ignore[attr-defined]
+        }
 
     @property
     def state(self):
         """Return the state of the device."""
         return self._state
 
+    @property
+    def code_arm_required(self):
+        """Whether the code is required for arm actions."""
+        return False
+
     @property
     def supported_features(self) -> int:
         """Return the list of supported features."""
         return SUPPORT_ALARM_ARM_HOME | SUPPORT_ALARM_ARM_AWAY
 
-    def update(self):
-        """Return the state of the device."""
-        armed_status = self._client.get_armed_status()
-
-        self._state = self._state_map.get(armed_status)
-
     def alarm_disarm(self, code=None):
         """Send disarm command."""
-        self._client.disarm()
+        self.coordinator.yale.disarm()
 
     def alarm_arm_home(self, code=None):
         """Send arm home command."""
-        self._client.arm_partial()
+        self.coordinator.yale.arm_partial()
 
     def alarm_arm_away(self, code=None):
         """Send arm away command."""
-        self._client.arm_full()
+        self.coordinator.yale.arm_full()
+
+    @property
+    def state_attributes(self):
```

Comment:
```suggestion
    def extra_state_attributes(self):
```

Reply:
This section has been removed completely